packages/bento-design-system/src/BentoProvider.tsx

```diff
@@ -38,16 +38,28 @@ type Props = {
   linkComponent?: ComponentType<LinkComponentProps>;
   locale?: string;
   config?: PartialBentoConfig;
-  theme?: BentoThemeOverride;
+  /**
+   * The theme to apply to the children.
+   * If not provided, the defaultTheme will be used.
+   * If explicitly set to `null`, no theme wrapper will be applied. This is useful when you want to
+   * manually load your global theme, e.g. created with vanilla-extract's createGlobalTheme.
+   */
+  theme?: BentoTheme | null;
   sprinkles?: SprinklesFn;
   ssr?: boolean;
 } & DefaultMessages;
 
 export function createBentoProvider(
   config: PartialBentoConfig = {},
-  theme?: BentoThemeOverride,
+  theme?: BentoTheme | null,
   sprinkles: SprinklesFn = bentoSprinkles
 ) {
+  function OptionalThemeWrapper(props: { children: Children; theme?: BentoTheme | null }) {
+    if (theme === null) return <>{props.children}</>;
+    const _theme = props.theme ?? defaultTheme;
+    return <BentoThemeProvider theme={_theme || defaultTheme}>{props.children}</BentoThemeProvider>;
```

Comment:
I'm a bit confused by all the defaulting here. Why `_theme || defaultTheme` if  `_theme` is already falling back to `defaultTheme`. Maybe just:

```suggestion
    const _theme = props.theme ?? defaultTheme;
    return <BentoThemeProvider theme={_theme}>{props.children}</BentoThemeProvider>;
```

?